cura/Settings/MachineManager.py

```diff
@@ -1314,14 +1314,14 @@ def updateMaterialWithVariant(self, position: Optional[str] = None) -> None:
         """
         if self._global_container_stack is None:
             return
-        if position is None:
-            position_list = [str(position) for position in range(len(self._global_container_stack.extruderList))]
-        else:
+        if position:
             position_list = [position]
+        else:
+            position_list = [str(position) for position, _ in enumerate(self._global_container_stack.extruderList)]
 
         for position_item in position_list:
             try:
-                extruder = self._global_container_stack.extruderList[int(position_item)]
+                extruder = self._global_container_stack.extruderList[int(position_)]
```

Comment:
```suggestion
                extruder = self._global_container_stack.extruderList[int(position_item)]
```

You changed the name of one of these position\_ -> position\_item, but you missed the rest of them. 

Please run the code or tests as well to make sure everything is working.